<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>PythonPluginCreationDialog</class>
 <widget class="QDialog" name="PythonPluginCreationDialog">
  <property name="windowModality">
   <enum>Qt::WindowModal</enum>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>574</width>
    <height>582</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Tulip Python plugin creation</string>
  </property>
  <property name="styleSheet">
   <string notr="true">background-color:white;</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QLabel" name="label_10">
     <property name="maximumSize">
      <size>
       <width>16777215</width>
       <height>40</height>
      </size>
     </property>
     <property name="text">
      <string>Please provide the required fields in order to generate the source code
skeleton of the Tulip Python plugin to create.</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox">
     <property name="title">
      <string>Required fields</string>
     </property>
     <layout class="QFormLayout" name="formLayout_2">
      <property name="fieldGrowthPolicy">
       <enum>QFormLayout::ExpandingFieldsGrow</enum>
      </property>
      <item row="0" column="0">
       <widget class="QLabel" name="label">
        <property name="toolTip">
         <string>the path to the file where the plugin source code will be saved</string>
        </property>
        <property name="text">
         <string>Plugin file: </string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <widget class="QLineEdit" name="pluginFileName"/>
        </item>
        <item>
         <widget class="QPushButton" name="browseButton">
          <property name="styleSheet">
           <string notr="true">.QWidget, QScrollArea, .QFrame {
background-color: white;
}</string>
          </property>
          <property name="text">
           <string/>
          </property>
          <property name="icon">
           <iconset resource="../resources/GraphPerspective.qrc">
            <normaloff>:/tulip/graphperspective/icons/16/document-open.png</normaloff>:/tulip/graphperspective/icons/16/document-open.png</iconset>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_3">
        <property name="toolTip">
         <string>The type of plugin to create</string>
        </property>
        <property name="text">
         <string>Plugin type: </string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QComboBox" name="pluginType">
        <item>
         <property name="text">
          <string>General</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Layout</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Size</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Measure</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Color</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Selection</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Import</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Export</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="label_2">
        <property name="toolTip">
         <string>The name of the Python plugin class</string>
        </property>
        <property name="text">
         <string>Plugin class name:</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QLineEdit" name="pluginClassName"/>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="label_4">
        <property name="toolTip">
         <string>The name of the plugin as it will appear in the interface</string>
        </property>
        <property name="text">
         <string>Plugin name:</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QLineEdit" name="pluginName"/>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox_2">
     <property name="title">
      <string>Optional fields</string>
     </property>
     <layout class="QFormLayout" name="formLayout">
      <item row="0" column="0">
       <widget class="QLabel" name="label_5">
        <property name="toolTip">
         <string>The name of the plugin's author</string>
        </property>
        <property name="text">
         <string>Author:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLineEdit" name="author"/>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_6">
        <property name="toolTip">
         <string>The creation date of the plugin</string>
        </property>
        <property name="text">
         <string>Date:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QLineEdit" name="date"/>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="label_7">
        <property name="toolTip">
         <string>Some information about the plugin</string>
        </property>
        <property name="text">
         <string>Information:</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QLineEdit" name="info"/>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="label_8">
        <property name="toolTip">
         <string>The version number of the plugin</string>
        </property>
        <property name="text">
         <string>Release:</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QLineEdit" name="release">
        <property name="text">
         <string>1.0</string>
        </property>
       </widget>
      </item>
      <item row="4" column="0">
       <widget class="QLabel" name="label_9">
        <property name="toolTip">
         <string>The plugin can be added to a group (will appear in a sub-menu in the interface)</string>
        </property>
        <property name="text">
         <string>Group: </string>
        </property>
       </widget>
      </item>
      <item row="4" column="1">
       <widget class="QLineEdit" name="group"/>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources>
  <include location="../resources/GraphPerspective.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>PythonPluginCreationDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>286</x>
     <y>556</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>290</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>PythonPluginCreationDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>286</x>
     <y>556</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>290</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
